internal/cmd/beta/kms/key/create/create.go

```diff
@@ -0,0 +1,210 @@
+		Protection:      flags.FlagToStringPointer(p, cmd, protectionFlag),
+	}
+
+	p.DebugInputModel(model)
+	return &model, nil
+}
+
+type kmsKeyClient interface {
+	CreateKey(ctx context.Context, projectId string, regionId string, keyRingId string) kms.ApiCreateKeyRequest
+}
+
+func buildRequest(ctx context.Context, model *inputModel, apiClient kmsKeyClient) (kms.ApiCreateKeyRequest, error) {
+	req := apiClient.CreateKey(ctx, model.ProjectId, model.Region, model.KeyRingId)
+
+	req = req.CreateKeyPayload(kms.CreateKeyPayload{
+		DisplayName: model.Name,
+		Description: model.Description,
+		Algorithm:   kms.CreateKeyPayloadGetAlgorithmAttributeType(model.Algorithm),
+		Purpose:     kms.CreateKeyPayloadGetPurposeAttributeType(model.Purpose),
+		ImportOnly:  &model.ImportOnly,
+		Protection:  kms.CreateKeyPayloadGetProtectionAttributeType(model.Protection),
+	})
+	return req, nil
+}
+
+func outputResult(p *print.Printer, model *inputModel, resp *kms.Key) error {
+	if resp == nil {
+		return fmt.Errorf("response is nil")
+	}
+
+	switch model.OutputFormat {
+	case print.JSONOutputFormat:
+		details, err := json.MarshalIndent(resp, "", "  ")
+		if err != nil {
+			return fmt.Errorf("marshal KMS key: %w", err)
+		}
+		p.Outputln(string(details))
+
+	case print.YAMLOutputFormat:
+		details, err := yaml.MarshalWithOptions(resp, yaml.IndentSequence(true), yaml.UseJSONMarshaler())
+		if err != nil {
+			return fmt.Errorf("marshal KMS key: %w", err)
+		}
+		p.Outputln(string(details))
+
+	default:
+		operationState := "Created"
+		if model.Async {
+			operationState = "Triggered creation of"
+		}
+		p.Outputf("%s the KMS key '%s'. Key ID: %s\n", operationState, utils.PtrString(resp.DisplayName), utils.PtrString(resp.Id))
+	}
+	return nil
+}
+
+func configureFlags(cmd *cobra.Command) {
+	// Algorithm
+	var algorithmFlagOptions []string
+	for _, val := range kms.AllowedAlgorithmEnumValues {
+		algorithmFlagOptions = append(algorithmFlagOptions, string(val))
+	}
+	cmd.Flags().Var(flags.EnumFlag(false, string(defaultAlgorithm), algorithmFlagOptions...), algorithmFlag, fmt.Sprintf("En-/Decryption / signing algorithm. Possible values: %q", algorithmFlagOptions))
+
+	// Purpose
+	var purposeFlagOptions []string
+	for _, val := range kms.AllowedPurposeEnumValues {
+		purposeFlagOptions = append(purposeFlagOptions, string(val))
+	}
+	cmd.Flags().Var(flags.EnumFlag(false, string(defaultPurpose), purposeFlagOptions...), purposeFlag, fmt.Sprintf("Purpose of the key. Possible values: %q", purposeFlagOptions))
+
+	// Protection
+	var protectionFlagOptions []string
+	for _, val := range kms.AllowedProtectionEnumValues {
+		protectionFlagOptions = append(protectionFlagOptions, string(val))
+	}
+	cmd.Flags().Var(flags.EnumFlag(false, string(defaultProtection), protectionFlagOptions...), protectionFlag, fmt.Sprintf("The underlying system that is responsible for protecting the key material. Possible values: %q", purposeFlagOptions))
```

Comment:
Set an empty string as defaultValue for the EnumFlag
```suggestion
	cmd.Flags().Var(flags.EnumFlag(false, "", algorithmFlagOptions...), algorithmFlag, fmt.Sprintf("En-/Decryption / signing algorithm. Possible values: %q", algorithmFlagOptions))

	// Purpose
	var purposeFlagOptions []string
	for _, val := range kms.AllowedPurposeEnumValues {
		purposeFlagOptions = append(purposeFlagOptions, string(val))
	}
	cmd.Flags().Var(flags.EnumFlag(false, "", purposeFlagOptions...), purposeFlag, fmt.Sprintf("Purpose of the key. Possible values: %q", purposeFlagOptions))

	// Protection
	var protectionFlagOptions []string
	for _, val := range kms.AllowedProtectionEnumValues {
		protectionFlagOptions = append(protectionFlagOptions, string(val))
	}
	cmd.Flags().Var(flags.EnumFlag(false, "", protectionFlagOptions...), protectionFlag, fmt.Sprintf("The underlying system that is responsible for protecting the key material. Possible values: %q", purposeFlagOptions))
```

Also in the other places where this is used. As long no default value is set in the api itself, it shouldn't be set here

Reply:
Got you. DNS is a special exception and the only place where a default value is set.